tracer/randomize.py

```diff
@@ -0,0 +1,106 @@
+import abc
+import random
+import string
+from typing import List, NoReturn, Sequence, Union
+
+_Number = Union[float, int]
+
+
+class Randomizer(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def create(self) -> NoReturn:
+        raise NotImplementedError
+
+
+class Integer(Randomizer):
+    def __init__(self, min: int = 1, max: int = 9):
+        self._min = min
+        self._max = max
+
+    def create(self) -> int:
+        return random.randint(self._min, self._max)
+
+
+class Double(Randomizer):
+    def __init__(self, min: _Number = 0, max: _Number = 1):
+        self._min = min
+        self._max = max
+
+    def create(self) -> float:
+        return random.uniform(self._min, self._max)
+
+
+class String(Randomizer):
+    def __init__(self, length: int = 16, letters: Sequence[str] = string.ascii_lowercase):
+        self._length = length
+        self._letters = letters
+
+    def create(self) -> str:
+        text = random.choices(self._letters, k=self._length)
+        return "".join(text)
+
+
+class Array1D(Randomizer):
+    def __init__(self, N: int = 10, randomizer: Randomizer = Integer()):
+        self._N = N
+        self._randomizer = randomizer
+        self._sorted = False
+
+    def sorted(self, sorted: bool = True) -> "Array1D":
+        self._sorted = sorted
+        return self
+
+    def create(self) -> List:
+        array = [self._randomizer.create() for _ in range(self._N)]
+        if self._sorted:
+            array.sort()
+
+        return array
+
+
+class Array2D(Array1D):
+    def __init__(self, N: int = 10, M: int = 10, randomizer: Randomizer = Integer()):
+        super().__init__(N, randomizer)
+        self._M = M
+
+    def sorted(self, sorted: bool = True) -> "Array2D":
```

Comment:
These sorts of setter functions are unidiomatic in Python. Ideally this should just be a normal instance attribute. However, this would be inconsistent with the other tracers because 1. it wouldn't be a function and 2. it wouldn't return `self`. Thoughts?

Reply:
I don't have much experience in Python, but I also felt the need for differing syntaxes of `Randomize` functions while writing the same functions in different languages. How would the usage look if it is a normal instance attribute? Is it a rare case to use method chaining to create a factory in Python?

Reply:
Yeah, I'd say it's a rare case but not out of the question (I have some speculation as to why; if you want to know more let me know). This setter is redundant because it doesn't have any logic in addition to setting the variable. One could argue the setter at least has some value in that it allows for chaining methods, but this could be replaced by adding it as an argument in the constructor:

```python
Array2D(sorted=True).create()  # Can also be used as a positional argument of course
```

Using an instance attribute instead:
```python
array2d = Array2D()
array2d.sorted = True
array2d.create()
```
However, naming it `sorted` would conflict with the function's name. 

Constructor + setter function would compromise nothing but constructor + instance attribute would be the most idiomatic. However, I do not think the latter outweighs the inconsistencies (with the other tracers) it would introduce. 

Another reason I am willing to concede the instance attribute is that it doesn't seem to me that its value would end up being changed often, if at all, during the lifetime of a randomiser instance. Consider that the randomisers as a whole weren't designed with this in mind anyway given that some attributes are private when they really could be public and not break anything. If we wanted something that can be configured easily between calls then at that point the best thing to do would be to drop the class shells and just have the individual `create()` functions take all the arguments.

Reply:
Ahh, I just realized that Python supports both positional and named arguments. Using a normal instance attribute makes more sense to me now. Actually, I agree that we should simply make randomizers functions, not classes, as I haven't had any case where I reuse randomizer instance with different options. Thus, the usage would be like: 

```py
Randomizer.createArray2D(sorted=True)
Randomizer.createGraph(N=5, directed=True)
```

I think JavaScript randomizers can follow the similar pattern as well (e.g., `Randomizer.createArray2D({sorted: true})`). I don't really come up with a similar solution for C++ and Java at this moment, so we can keep them using class and method chaining for now.

Reply:
Yeah, this discussion went out of the scope of this PR. For now I'll just stay close to the API and maybe later PRs could be made to redesign some things. But you do make a good point about how to support this in C++ and Java; I don't think a billion constructor overloads are practical.